### ISSUE TITLE
Can't build next under debian wheezy

### ISSUE THREAD

**Issue by intuser** [atlas-engineer/next#138]
As the installation under ubuntu 18.04 didn't work out for me, I tried another install under debian wheezy. Here a problem arises even earlier (sbcl is SBCL 1.2.4.debian):

(asdf:load-asd "/full/path/to/next.asd") results in:

```debugger invoked on a SB-INT:SIMPLE-READER-PACKAGE-ERROR in thread
#<THREAD "main thread" RUNNING {10039CE5A3}>:
  The symbol "LOAD-ASD" is not external in the ASDF/INTERFACE package.

    Stream: #<SYNONYM-STREAM :SYMBOL SB-SYS:*STDIN* {10001C21C3}>

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

 restarts (invokable by number or by possibly-abbreviated name):
  0: [CONTINUE] Use symbol anyway.
  1: [ABORT   ] Exit debugger, returning to top level.

 (SB-IMPL::READ-TOKEN #<SYNONYM-STREAM :SYMBOL SB-SYS:*STDIN* {10001C21C3}> #\a)
0]

**Comment by jmercouris** [comment]
Hi, thank you for your report!

What version of ASDF are you running? If the symbol is not external you can always try 

(asdf::load-asd "/full/path/to/next.asd")

**Comment by intuser** [comment]
2.26

With (asdf::load-asd "/full/path/to/next.asd") I get:

```
[package swank/sbcl].....; 
; caught ERROR:
;   READ error during COMPILE-FILE:
;   
;     The symbol "EARLY-DEPRECATION-WARNING" is not external in the SB-EXT package.
;   
;       Line: 523, Column: 55, File-Position: 18768
;   
;       Stream: #<SB-SYS:FD-STREAM
;                 for "file /home/xxx/quicklisp/dists/quicklisp/software/slime-v2.22/swank/sbcl.lisp"
;                 {1006B63873}>
;; 
;; Error compiling /home/xxx/quicklisp/dists/quicklisp/software/slime-v2.22/swank/sbcl.lisp:
;;   COMPILE-FILE returned NIL.
;; 

debugger invoked on a SIMPLE-ERROR in thread
#<THREAD "main thread" RUNNING {10039CE573}>:
  COMPILE-FILE returned NIL.
```

**Comment by vindarel** [comment]
2.26, built in Debian's SBCL, is a bit old. Upgrading is as simple as cloning asdf into `~/common-lisp/asdf`, can you try that ?

https://common-lisp.net/project/asdf/asdf.html#Upgrading-ASDF

    cd ~/common-lisp/ &&   git clone https://github.com/fare/asdf .

**Comment by intuser** [comment]
I did so  and I get the same error. Is there something I should delete before? I did delete the asdf.lisp file version 2.26. (After a first unsuccessful try.)

**Comment by vindarel** [comment]
mmh what about using Quicklisp ? `(ql:quickload :next-gtk)` ? (after you compiled `C-c C-k` the .asd)

**Comment by intuser** [comment]
"after you compiled C-c C-k the .asd"

Does that mean that I have to install and learn to use slime? Or can I compile the next.asd (I guess you are speaking about that) some other way?

Anyway, I tried to use "make" even though running next from the source was unsuccessful and I got the following result:

```
sbcl    --non-interactive \
        --eval '(require "asdf")' \
        --load next.asd \
        --eval '(ql:quickload :next/gtk)' \
        --eval '(setq *exit-timeout* nil)' \
        --eval '(asdf:make :next/gtk)'
This is SBCL 1.2.4.debian, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
To load "next/gtk":
  Load 1 ASDF system:
    next/gtk
; Loading "next/gtk"
...To load "lparallel":
  Load 2 ASDF systems:
    alexandria bordeaux-threads
  Install 1 Quicklisp release:
    lparallel
; Fetching #<URL "http://beta.quicklisp.org/archive/lparallel/2016-08-25/lparallel-20160825-git.tgz">
; 76.71KB
==================================================
78,551 bytes in 0.01 seconds (8523.33KB/sec)
; Loading "lparallel"
[package lparallel.util]..........................
[package lparallel.thread-util]...................
[package lparallel.raw-queue].....................
[package lparallel.cons-queue]....................
[package lparallel.vector-queue]..................
[package lparallel.queue].........................
[package lparallel.counter].......................
[package lparallel.spin-queue]....................
[package lparallel.kernel]........................
[package lparallel.kernel-util]...................
[package lparallel.promise].......................
[package lparallel.ptree].........................
[package lparallel.slet]..........................
[package lparallel.defpun]........................
[package lparallel.cognate].......................
[package lparallel]
; Loading "next/gtk"
..................................................
[package swank-loader]...Unhandled UNDEFINED-FUNCTION in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                          {10039CED03}>:
  The function ASDF/INTERFACE::OPERATION-FORCED is undefined.

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10039CED03}>
0: ("undefined function")
1: ((:METHOD ASDF/ACTION:PERFORM (ASDF/LISP-ACTION:LOAD-OP SWANK-LOADER::SWANK-LOADER-FILE)) #<ASDF/LISP-ACTION:LOAD-OP > #<SWANK-LOADER::SWANK-LOADER-FILE "swank" "swank-loader">) [fast-method]
2: ((SB-PCL::EMF ASDF/ACTION:PERFORM) #<unavailable argument> #<unavailable argument> #<ASDF/LISP-ACTION:LOAD-OP > #<SWANK-LOADER::SWANK-LOADER-FILE "swank" "swank-loader">)
3: ((LAMBDA NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
4: ((:METHOD ASDF/ACTION:PERFORM-WITH-RESTARTS (ASDF/LISP-ACTION:LOAD-OP ASDF/LISP-ACTION:CL-SOURCE-FILE)) #<ASDF/LISP-ACTION:LOAD-OP > #<SWANK-LOADER::SWANK-LOADER-FILE "swank" "swank-loader">) [fast-method]
5: ((:METHOD ASDF/ACTION:PERFORM-WITH-RESTARTS :AROUND (T T)) #<ASDF/LISP-ACTION:LOAD-OP > #<SWANK-LOADER::SWANK-LOADER-FILE "swank" "swank-loader">) [fast-method]
6: ((:METHOD ASDF/PLAN:PERFORM-PLAN (T)) #<ASDF/PLAN:SEQUENTIAL-PLAN {1007EB0133}>) [fast-method]
7: ((FLET SB-C::WITH-IT :IN SB-C::%WITH-COMPILATION-UNIT))
8: ((:METHOD ASDF/PLAN:PERFORM-PLAN :AROUND (T)) #<ASDF/PLAN:SEQUENTIAL-PLAN {1007EB0133}>) [fast-method]
9: ((:METHOD ASDF/OPERATE:OPERATE (ASDF/OPERATION:OPERATION ASDF/COMPONENT:COMPONENT)) #<ASDF/LISP-ACTION:LOAD-OP > #<ASDF/SYSTEM:SYSTEM "next/gtk"> :PLAN-CLASS NIL :PLAN-OPTIONS NIL) [fast-method]
10: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> #<ASDF/LISP-ACTION:LOAD-OP > #<ASDF/SYSTEM:SYSTEM "next/gtk"> :VERBOSE NIL)
11: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
12: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) #<ASDF/LISP-ACTION:LOAD-OP > #<ASDF/SYSTEM:SYSTEM "next/gtk"> :VERBOSE NIL) [fast-method]
13: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> ASDF/LISP-ACTION:LOAD-OP "next/gtk" :VERBOSE NIL)
14: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
15: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) ASDF/LISP-ACTION:LOAD-OP "next/gtk" :VERBOSE NIL) [fast-method]
16: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<CLOSURE (LAMBDA NIL :IN ASDF/OPERATE:OPERATE) {1007E9615B}> :OVERRIDE T :KEY NIL :OVERRIDE-CACHE T :OVERRIDE-FORCING NIL)
17: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
18: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<CLOSURE (LAMBDA NIL :IN ASDF/OPERATE:OPERATE) {1007DD411B}> :OVERRIDE NIL :KEY NIL :OVERRIDE-CACHE NIL :OVERRIDE-FORCING NIL)
19: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) ASDF/LISP-ACTION:LOAD-OP "next/gtk" :VERBOSE NIL) [fast-method]
20: (ASDF/OPERATE:LOAD-SYSTEM "next/gtk" :VERBOSE NIL)
21: (QUICKLISP-CLIENT::CALL-WITH-MACROEXPAND-PROGRESS #<CLOSURE (LAMBDA NIL :IN QUICKLISP-CLIENT::APPLY-LOAD-STRATEGY) {1007DD407B}>)
22: (QUICKLISP-CLIENT::AUTOLOAD-SYSTEM-AND-DEPENDENCIES "next/gtk" :PROMPT NIL)
23: ((:METHOD QL-IMPL-UTIL::%CALL-WITH-QUIET-COMPILATION (T T)) #<unavailable argument> #<CLOSURE (FLET QUICKLISP-CLIENT::QL :IN QUICKLISP-CLIENT:QUICKLOAD) {1005900F9B}>) [fast-method]
24: ((:METHOD QL-IMPL-UTIL::%CALL-WITH-QUIET-COMPILATION :AROUND (QL-IMPL:SBCL T)) #<QL-IMPL:SBCL {1004C72EC3}> #<CLOSURE (FLET QUICKLISP-CLIENT::QL :IN QUICKLISP-CLIENT:QUICKLOAD) {1005900F9B}>) [fast-method]
25: ((:METHOD QUICKLISP-CLIENT:QUICKLOAD (T)) #<unavailable argument> :PROMPT NIL :SILENT NIL :VERBOSE NIL) [fast-method]
26: (QL-DIST::CALL-WITH-CONSISTENT-DISTS #<CLOSURE (LAMBDA NIL :IN QUICKLISP-CLIENT:QUICKLOAD) {10058E9F7B}>)
27: (SB-INT:SIMPLE-EVAL-IN-LEXENV (QUICKLISP-CLIENT:QUICKLOAD :NEXT/GTK) #<NULL-LEXENV>)
28: (EVAL (QUICKLISP-CLIENT:QUICKLOAD :NEXT/GTK))
29: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:EVAL . "(require \"asdf\")") (:LOAD . "next.asd") (:EVAL . "(ql:quickload :next/gtk)") (:EVAL . "(setq *exit-timeout* nil)") (:EVAL . "(asdf:make :next/gtk)") (:QUIT)))
30: (SB-IMPL::TOPLEVEL-INIT)
31: ((FLET #:WITHOUT-INTERRUPTS-BODY-89 :IN SAVE-LISP-AND-DIE))
32: ((LABELS SB-IMPL::RESTART-LISP :IN SAVE-LISP-AND-DIE))

unhandled condition in --disable-debugger mode, quitting
; 
; compilation unit aborted
;   caught 1 fatal ERROR condition
Makefile:4: recipe for target 'build-gtk' failed
make: *** [build-gtk] Error 1
```

**Comment by vindarel** [comment]
ok, given you type lisp commands I was supposing you were in Slime, but no pb, `make` is good.

Sooo, are you really using ASDF 3 ? To find out, I suggest to add this in your .sbclrc:

    (format t "--- asdf version: ~a~&" (asdf:asdf-version))

Also, asking for these messages on Stack Overflow might help…

**Comment by intuser** [comment]
This results in:
--- asdf version: 3.3.1.2

**Comment by vindarel** [comment]
Well done then :)

I have the same stack, no more idea :(

**Comment by jmercouris** [comment]
@intuser If you are just willing to wait a few more weeks, or donate to our campaign ;) (https://www.indiegogo.com/projects/next-browser-nix-support) hopefully all of your compilation woes will be solved :D

**Comment by Ambrevar** [comment]
This should now be fixed. @intuser can you confirm?

**Comment by intuser** [comment]
Unfortunately it isn't fixed and actually there's an additional error.

Please have a look at https://pastebin.com/qiuTvfra

**Comment by Ambrevar** [comment]
You are using SBCL 1.2.4, which is very old and its ASDF implementation seems unsupported by SWANK.

Updating to SBCL to 1.4+ will fix your issue.

**Comment by Ambrevar** [comment]
Also see https://www.common-lisp.net/project/asdf/asdf.html#Upgrading-ASDF.
@vindarel What about you?

**Comment by intuser** [comment]
The old errors went away, but there's a new one:

https://pastebin.com/L6PzLeug

**Comment by intuser** [comment]
$ pkg-config --cflags webkit2gtk-4.0
-pthread -I/usr/include/webkitgtk-4.0 -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng12 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng12 -I/usr/include/libsoup-2.4 -I/usr/include/libxml2 -I/usr/include/webkitgtk-4.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include

libsoup is 2.48.0-1

**Comment by Ambrevar** [comment]
I think you need to install libsoup2.4-dev (could be a packaging issue on Debian's side).
There might be a few more like this you'll need on Debian-based system.

Let me know if there are more -dev packages to install and I'll update the README accordingly.

**Comment by intuser** [comment]
I have libsoup2.4-dev already installed. How could I find out what others might be needed?

**Comment by Ambrevar** [comment]
Sorry, I had misread the log.  The issue is with your C compiler:

> window.h:141:2: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
>   for (int i = 0; i < (sizeof key_blacklist)/(sizeof key_blacklist[0]); i++) {
>   ^
> window.h:141:2: note: use option -std=c99, -std=gnu99, -std=c11 or -std=gnu11 to compile your code

It looks like your C compiler forces to C89 by default.
Do you have any exported CFLAGS?  What's the output of `echo $CFLAGS`?
Try running `make gtk-webkit CFLAGS+=-stdc11` and see if this works.

**Comment by intuser** [comment]
First try:

$ make gtk-webkit CFLAGS+=-stdc11
make -C ports/gtk-webkit
make[1]: Entering directory '/home/andreas/next-master/ports/gtk-webkit'
cc -stdc11   -c -o next-gtk-webkit.o next-gtk-webkit.c
cc: error: unrecognized command line option ‘-stdc11’

Second try:

$ make gtk-webkit CFLAGS+=-std=c11
make -C ports/gtk-webkit
make[1]: Entering directory '/home/andreas/next-master/ports/gtk-webkit'
cc -std=c11   -c -o next-gtk-webkit.o next-gtk-webkit.c
next-gtk-webkit.c:6:21: fatal error: gtk/gtk.h: No such file or directory
 #include <gtk/gtk.h>

Third try (with -std=c11 in the makefile): https://pastebin.com/9j7zGPHy

It seems SoupXMLRPCParams was only introduced in libsoup 2.52 (https://developer.gnome.org/libsoup/stable/libsoup-2.4-XMLRPC-Support.html#SoupXMLRPCParams) and next-gtk cannot be compiled on debian oldstable.

**Comment by intuser** [comment]
Debian oldstable (jessie) still receives security support.

**Comment by Ambrevar** [comment]
Sorry, but I don't know what else to suggest.  Libsoup 2.52 was released more
than 3 years ago, and it's a central piece of the GTK-WebKit port.

About the C standard: I gave you the wrong command for `make`.  When variables
are set as argument, they override the Makefile's definition.  The proper way to
append a CFLAG is to set the variable in the environment, before calling make.
For instance:

```
env CFLAGS=-std=c11 make gtk-webkit
```
